blobstore/src/test/java/org/jclouds/blobstore/integration/internal/BaseContainerLiveTest.java

```diff
@@ -112,12 +116,46 @@ public void testPublicAccessInNonDefaultLocationWithBigBlob() throws Interrupted
       runCreateContainerInLocation(payload, nonDefault);
    }
 
+   @Test(groups = "live", dependsOnMethods = "testPublicAccess")
+   public void testDelimiterList() throws InterruptedException {
+      final String containerName = getScratchContainerName();
+      BlobStore blobStore = view.getBlobStore();
+      String payload = "foo";
+      try {
+         createContainer(containerName, defaultLocation);
+         blobStore.putBlob(containerName, blobStore.blobBuilder("test/foo/foo").payload(payload).build());
+         blobStore.putBlob(containerName, blobStore.blobBuilder("test/bar/foo").payload(payload).build());
+         blobStore.putBlob(containerName, blobStore.blobBuilder("foo").payload(payload).build());
+         checkListedSet(ImmutableSet.of("foo", "test"), blobStore.list(
+                 containerName, ListContainerOptions.Builder.delimiter('/')));
+         checkListedSet(ImmutableSet.of("test/foo/foo", "test/bar/foo", "foo"), blobStore.list(
+                 containerName, ListContainerOptions.Builder.delimiter('\\')));
+         checkListedSet(ImmutableSet.of("test/foo", "test/bar"), blobStore.list(
+                 containerName, ListContainerOptions.Builder.delimiter('/').inDirectory("test")));
+         checkListedSet(ImmutableSet.of("test/bar/foo"), blobStore.list(
+                 containerName, ListContainerOptions.Builder.delimiter('/').inDirectory("test/bar")));
+      } finally {
+         recycleContainerAndAddToPool(containerName);
+      }
+   }
+
+   private void checkListedSet(ImmutableSet<String> expectedSet, PageSet<? extends StorageMetadata> results) {
+      Set<String> resultSet = new HashSet<String>();
+      for (StorageMetadata sm : results) {
+         if (!resultSet.contains(sm.getName())) {
+            resultSet.add(sm.getName());
+         }
+      }
+      assertEquals(resultSet, expectedSet);
+   }
```

Comment:
Would this be more clear as a `storageMetadataToNames` method combined with `assertThat(Iterable).containsExactly(Iterable)`?